### DISCUSSION TITLE
Tunable retry count for GLPI::Agent::SNMP::Live

### DISCUSSION THREAD

**Discussion by po1vo** [glpi-project/glpi-agent#642] [Ideas]
### Your idea

I suggest making ```-retries``` option from Net::SNMP available. Let's call it snmp_retries and add to agent.conf

The reason is that some of requests to some of our equipment time out in the middle of response or don't respond at all. This results in incomplete and incorrect data, like missing MACs on all interfaces, which creates chaos inside GLPI.

The idea is a workaround, but at least it's something.

**Comment by g-bougard** [comment]
Hi @po1vo 

I understand your concern.

Did you try to change hard-coded value [here](https://github.com/glpi-project/glpi-agent/blob/0a77a793dc2c254e83d0cff5e9201fbf86b2b97a/lib/GLPI/Agent/SNMP/Live.pm#L29) and [here](https://github.com/glpi-project/glpi-agent/blob/0a77a793dc2c254e83d0cff5e9201fbf86b2b97a/lib/GLPI/Agent/SNMP/Live.pm#L102) from `0` to a positive value ?

**Reply by po1vo** [reply]
Yes, and this fixed the issue for me.

**Reply by g-bougard** [reply]
Hi @po1vo 

I just integrated your feature request in glpi-agent. Can I ask you to test it with next nightly build ?

The parameter is `snmp-retries` in configuration. It will be also supported by using `--retries` option with `glpi-netdiscovery` and `glpi-netinventory` scripts.
